rs/execution_environment/src/scheduler/tests.rs

```diff
@@ -1699,21 +1699,15 @@ fn can_fully_execute_multiple_canisters_with_multiple_messages_each() {
     for canister_state in test.state().canisters_iter() {
         let system_state = &canister_state.system_state;
         assert_eq!(system_state.queues().ingress_queue_size(), 0);
-        assert_eq!(
-            canister_state.scheduler_state.last_full_execution_round,
-            ExecutionRound::new(1)
-        );
-        assert_eq!(
-            system_state
-                .canister_metrics
-                .skipped_round_due_to_no_messages,
-            0
-        );
-        assert_eq!(system_state.canister_metrics.executed, 1);
-        assert_eq!(
-            system_state.canister_metrics.interrupted_during_execution,
-            0
-        );
+
+        let scheduler_state = &canister_state.scheduler_state;
+        assert_eq!(scheduler_state.last_full_execution_round, test.last_round());
+
+        let metrics = &system_state.canister_metrics;
+        // The inner round was skipped once before breaking the round.
+        assert_eq!(metrics.skipped_round_due_to_no_messages, 1);
```

Comment:
Low priority: Could we avoid unnecessarily bumping this counter on every round when we execute all messages?

Or rename it to something like "rounds when we executed all messages"? (Although this can probably be inferred from the number of instructions executed in that round.)

Reply:
Sure, we'll have performance optimizations in https://dfinity.atlassian.net/browse/EXC-1617 It's an orthogonal change to this MR.